src/quicklist.c

```diff
@@ -1485,19 +1484,13 @@ void quicklistRotate(quicklist *quicklist) {
     char longstr[32] = {0};
 
     if(unlikely(quicklist->tail->container == QUICKLIST_NODE_CONTAINER_NONE)) {
-
-        quicklistNode *head = quicklist->tail;
-        quicklistNode *tail = quicklist->head;
-
-        head->next = quicklist->head->next;
-        head->prev = NULL;
-
-        tail->next = NULL;
-        tail->prev = quicklist->tail->prev;
-
-        quicklist->head = head;
-        quicklist->tail = tail;
-
+        quicklistNode *new_head = quicklist->tail;
+        quicklistNode *new_tail = quicklist->tail->prev;
+        new_tail->next = NULL;
+        new_head->next = quicklist->head->next;
+        new_head->prev = NULL;
+        quicklist->head->next->prev = new_head;
+        quicklist->head = new_head;
```

Comment:
Not sure if I'm understanding this correctly, is this the way it should be?
```c
        new_head->next = quicklist->head;
        new_head->prev = NULL;
        quicklist->head->prev = new_head;
        quicklist->head = new_head;
```

Reply:
I committed by mistake... I will commit it now with the test...